### ISSUE TITLE
prometheus_multiproc_dir is not cleaned on container restart

### ISSUE THREAD

**Issue by higels** [nautobot/nautobot#4234]
### Environment
* Nautobot version (Docker tag too if applicable): 9d4fd1b2833a (v1.2.11) 
* Python version: 3.8.10
* Database platform, version: postgres 12
* Middleware(s): n/a

### Steps to Reproduce
1. Start the nautobot container
2. Leave it running so that it generates some prometheus metric db files in /prom_cache
3. Restart the docker container

### Expected Behavior
I would have expected the old prometheus muiltiprocessing db files to be cleaned up when the container is restarted.

### Observed Behavior
Instead, they persist on disk, and with a large number of processes, can amount to hundreds of thousands of files over the course of a few months and eventually run the filesystem out of inodes.

We are running quite an old version, but it looks like this behaviour is still present in head.`docker-entrypoint.sh` seems like the right place to perform this clean-up, per the instructions in the client library:

> The PROMETHEUS_MULTIPROC_DIR environment variable must be set to a directory that the client library can use for metrics. This directory must be wiped between process/Gunicorn runs (before startup is recommended).

**Comment by bryanculver** [comment]
Related: #2455

**Comment by glennmatthews** [comment]
Does the same problem manifest if the container never gets restarted? Is there something we need to configure on prometheus itself to do automated cleanup of older files?

**Comment by higels** [comment]
Yes, it certainly looks that way. Our current instance has been up for 14 days, and has produced 80K db files.

I've dug a bit more, and it appears to be the health check processes that are generating all of these files.

Each UWSGI worker just has handles open to a single set of consistently numbered db files:

```
nautobot- 2457 systemd-coredump    3u      REG   0,54    65536 268911 /prom_cache/counter_1.db
nautobot- 2457 systemd-coredump    4u      REG   0,54    65536 268911 /prom_cache/counter_1.db
nautobot- 2457 systemd-coredump    5u      REG   0,54    65536 678649 /prom_cache/gauge_max_1.db
nautobot- 2457 systemd-coredump    6u      REG   0,54    65536 678649 /prom_cache/gauge_max_1.db
nautobot- 2457 systemd-coredump    7u      REG   0,54    65536 678650 /prom_cache/histogram_1.db
nautobot- 2457 systemd-coredump   19u      REG   0,54    65536 678650 /prom_cache/histogram_1.db
```

(which is good).

But if I look at a more recently created counter and gauge file in there:

```
nautobot@c0224afc836f:~$ ls /prom_cache/*1754135*
/prom_cache/counter_1754135.db  /prom_cache/gauge_max_1754135.db

nautobot@c0224afc836f:~$ cat /prom_cache/*1754135*
@N["cacheops_cache_hit", "cacheops_cache_hit_total", {}, "Number of cache hits"]      R["cacheops_cache_miss", "cacheops_cache_miss_total", {}, "Number of cache misses"]  g["cacheops_cache_invalidated", "cacheops_cache_invalidated_total", {}, "Number of cache invalidations"]     ]["health_check_database_info", "health_check_database_info", {}, "State of database backend"]       ?d["health_check_redis_backend_info", "health_check_redis_backend_info", {}, "State of redis backend"]
```

These all seem to be the result of health check runs. My understanding of this stuff is rusty, but since we should only have one health check running at a time, MultiProcessCollector might not be the right choice for the health checks specifically.
